### PR TITLE
Associate image with the right project on the joined node

### PR DESCRIPTION
**Pull request by adglkh**

### CODE REVIEW

File: lxd/api_cluster.go [hunk 1]
```diff
@@ -553,8 +553,17 @@ func clusterPutJoin(d *Daemon, req api.ClusterPut) Response {
 					return err
 				}
 
-				project := "default"
-				return d.cluster.ImageAssociateNode(project, fingerprint)
+				projects, err := d.cluster.ImagesGetProjectsWithImage(fingerprint)
+				if err != nil {
+					return err
+				}
```
Comment (stgraber) [inline lxd/api_cluster.go]: Line break please

File: lxd/api_cluster.go [hunk 2]
```diff
@@ -553,8 +553,17 @@ func clusterPutJoin(d *Daemon, req api.ClusterPut) Response {
...
+				for _, project := range projects {
+					err := d.cluster.ImageAssociateNode(project, fingerprint)
+					if err != nil {
+						return err
+					}
+				}
```
Comment (stgraber) [inline lxd/api_cluster.go]: Line break please

File: lxd/db/images.go [hunk 3]
```diff
@@ -912,6 +912,22 @@ func (c *Cluster) ImagesGetByNodeID(id int64) ([]string, error) {
 	return addresses, err
 }
 
+// ImagesGetProjectsWithImage returns all projects name that the image associated with the given fingerprint.
+func (c *Cluster) ImagesGetProjectsWithImage(fingerprint string) ([]string, error) {
```
Comment (stgraber) [inline lxd/db/images.go]: Why the new function and repetitive calls?

Couldn't you modify your ImagesGetByNodeID to return a map[string][]string{} that would give you a list of project for a fingerprint? (key is fingerprint, value is list of projects)

**Comment by adglkh** [comment]
Updated. Please take another look.
Thanks!

**Comment by stgraber** [comment]
jenkins: test this please

**Comment by stgraber** [comment]
Haven't re-reviewed yet but let's have Jenkins do a run anyway.

**Comment by stgraber** [comment]
autoSyncImagesTask should be a cluster task, not a global one

**Comment by stgraber** [comment]
Looks good, I'll do a quick cleanup for what I mentioned above, run it through Jenkins again then merge.

**Comment by stgraber** [comment]
jenkins: test this please

**Comment by stgraber** [comment]
jenkins: test this please

**Pull request merged**